cmd/notation/internal/truststore/truststore.go

```diff
@@ -0,0 +1,226 @@
+	x509.ExtKeyUsageIPSECEndSystem:                 "IPSec End System",
+	x509.ExtKeyUsageIPSECTunnel:                    "IPSec Tunnel",
+	x509.ExtKeyUsageIPSECUser:                      "IPSec User",
+	x509.ExtKeyUsageTimeStamping:                   "Time Stamping",
+	x509.ExtKeyUsageOCSPSigning:                    "OCSP Signing",
+	x509.ExtKeyUsageMicrosoftServerGatedCrypto:     "Microsoft Server Gated Crypto",
+	x509.ExtKeyUsageNetscapeServerGatedCrypto:      "Netscape Server Gated Crypto",
+	x509.ExtKeyUsageMicrosoftCommercialCodeSigning: "Microsoft Commercial Code Signing",
+	x509.ExtKeyUsageMicrosoftKernelCodeSigning:     "Microsoft Kernel Code Signing",
+}
+
+// AddCert adds a single cert file at path to the trust store
+// under dir truststore/x509/storeType/namedStore
+func AddCert(path, storeType, namedStore string, display bool) error {
+	// initialize
+	certPath, err := filepath.Abs(path)
+	if err != nil {
+		return err
+	}
+	if storeType == "" {
+		return errors.New("store type cannot be empty")
+	}
+	if !ValidateStoreType(storeType) {
+		return fmt.Errorf("unsupported store type: %s", storeType)
+	}
+	if namedStore == "" {
+		return errors.New("named store cannot be empty")
+	}
+
+	// check if the target path is a cert (support PEM and DER formats)
+	if _, err := corex509.ReadCertificateFile(certPath); err != nil {
+		return err
+	}
+
+	// core process
+	// get the trust store path
+	trustStorePath, err := dir.Path.UserConfigFS.GetPath(dir.TrustStoreDir, "x509", storeType, namedStore)
+	if err := CheckNonErrNotExistError(err); err != nil {
+		return err
+	}
+	// check if certificate already in the trust store
+	if _, err := os.Stat(filepath.Join(trustStorePath, filepath.Base(certPath))); err == nil {
+		return errors.New("certificate already exists in the Trust Store")
+	}
+	// add cert to trust store
+	_, err = osutil.CopyToDir(certPath, trustStorePath)
+	if err != nil {
+		return err
+	}
+
+	// write out
+	if display {
+		fmt.Printf("Successfully added %s to named store %s of type %s\n", filepath.Base(certPath), namedStore, storeType)
+	}
+
+	return nil
+}
+
+// ListCerts walks through root and lists all x509 certificates in it,
+// sub-dirs are ignored.
+func ListCerts(root string, depth int) error {
+	maxDepth := strings.Count(root, string(os.PathSeparator)) + depth
+
+	return filepath.WalkDir(root, func(path string, d fs.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
+		if d.IsDir() && strings.Count(path, string(os.PathSeparator)) > maxDepth {
+			return fs.SkipDir
+		}
+		info, err := d.Info()
+		if err != nil {
+			return err
+		}
+		if info.Mode().IsRegular() {
+			if _, err := corex509.ReadCertificateFile(path); err != nil {
+				return err
+			}
+			fmt.Println(path)
+		}
+		return nil
+	})
+}
+
+// ShowCerts writes out details of certificates
+func ShowCerts(certs []*x509.Certificate) {
+	fmt.Println("Display certificate details. Starting from leaf certificate if it's a certificate chain.")
```

Comment:
We are not validating the `certs` array is a certificate chain.
```suggestion
	fmt.Println("Certificate details")
```